lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2797,17 +2797,39 @@ void ConvertLayoutOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
 void LocalAllocOp::getEffects(
     SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
         &effects) {
-  // For alloc that cannot be mutated we mark them as no-side effect. This is
```

Comment:
Hi @ThomasRaoux , may I know why we marked it as no side effect?

Reply:
ah right, this was to allow things like CSE, DCE to work on those. The idea was that if the memory is immutable technically its side effects are not visible to other ops. This is not really true once we create allocation offsets.

Reply:
one "hacky" solution would be to check if allocation.offset is set and then have side-effects in this case?